pkg/apis/nsx.vmware.com/v1alpha1/ipaddressallocation_types.go

```diff
@@ -0,0 +1,58 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
+
+// IPAddressAllocation is the Schema for the ipallocation API.
+// +kubebuilder:printcolumn:name="IPAddressBlockVisibility",type=string,JSONPath=`.spec.ip_address_block_visibility`,description="IPAddressBlockVisibility of IPAddressAllocation"
+// +kubebuilder:printcolumn:name="CIDR",type=string,JSONPath=`.status..cidr`,description="CIDRs for the IPAddressAllocation"
+type IPAddressAllocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   IPAddressAllocationSpec   `json:"spec"`
+	Status IPAddressAllocationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// IPAddressAllocationList contains a list of IPAddressAllocation.
+type IPAddressAllocationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IPAddressAllocation `json:"items"`
+}
+
+// IPAddressAllocationSpec defines the desired state of IPAddressAllocation.
+type IPAddressAllocationSpec struct {
+	// IPAddressBlockVisibility defines the visibility of IP address block, External or Private.
```

Comment:
IPAddressBlockVisibility specified the visibility of the IPBlocks to allocate IP addresses. Can be External or Private.

Reply:
ok